aten/src/THCUNN/MultiLabelMarginCriterion.cu

```diff
@@ -133,6 +140,11 @@ __global__ void cunn_MultiLabelMarginCriterion_updateGradInput_kernel(Dtype *gra
     }
     __syncthreads();
```

Comment:
Are this syncthreads and the one three lines below necessary?

Reply:
I think you're right about the one below. I'll take that out. Correct me if I'm wrong, but isn't this one needed so that sum doesn't change on other threads when the reduce is being run?

Reply:
On second thought, I think it's necessary so that `gradInput_k[target_idx]` isn't being written to by multiple threads.

Reply:
@li-roy I think the syncthreads before the reduction makes sure that data is available at all threads when doing reduction.

@zou3519 I think only thread 0 writes to `gradInput_k[target_idx]`

Reply:
@SsnL But the syncthreads after the reduction is needed so that each thread doesn't continue to the next iteration of the loop and overwrite sum before the reduction is completed on thread 0.

Reply:
Yes, `sum` may be overwritten. But in `reduceBlock`, the values are first into shared memory (`sums` here) and a `__syncthreads()` is called (see https://github.com/pytorch/pytorch/blob/502aaf39cf4a878f9e4f849e5f409573aa598aa9/aten/src/THC/THCReduceApplyUtils.cuh#L39-L48). Since there is `__syncthreads()` before the reduction, the `sums` shared memory won't be overwritten if I'm understanding correctly.

Reply:
Oh okay, in that case I agree. Thanks for the pointer!